pallets/auctions/src/types/candle.rs

```diff
@@ -0,0 +1,277 @@
+						dest,
+					)?;
+
+					self.specific_data.winner = Some(winner.clone());
+
+					let auction_account = &Pallet::<T>::get_auction_subaccount_id(auction_id);
+					let auction_balance = <<T as crate::Config>::Currency as Currency<T::AccountId>>::free_balance(auction_account);
+					let reserved_amount = <ReservedAmounts<T>>::get(&winner, &auction_id);
+
+					ensure!(
+						reserved_amount > Zero::zero(),
+						Error::<T>::NoReservedAmountAvailableToClaim
+					);
+
+					<<T as crate::Config>::Currency as Currency<T::AccountId>>::transfer(
+						auction_account,
+						&self.common_data.owner,
+						reserved_amount,
+						ExistenceRequirement::AllowDeath,
+					)?;
+
+					<ReservedAmounts<T>>::remove(&winner, &auction_id);
+
+					// Only one bidder: Auction and related data can be pruned
+					if reserved_amount == auction_balance {
+						destroy_auction_data = true;
+					}
+				}
+				None => return Err(Error::<T>::ErrorDeterminingAuctionWinner.into()),
+			}
+		} else {
+			destroy_auction_data = true;
+		}
+
+		Ok(destroy_auction_data)
+	}
+
+	///
+	/// Claims reserved amounts which were bid on a Candle auction
+	/// 
+	fn claim(&self, auction_id: T::AuctionId, bidder: T::AccountId, amount: BalanceOf<T>) -> Result<bool, DispatchError> {
+		let mut destroy_auction_data = false;
+
+		Pallet::<T>::validate_claim(&self.common_data)?;
+		Pallet::<T>::handle_claim(bidder, auction_id, amount)?;
+
+		let auction_account = &Pallet::<T>::get_auction_subaccount_id(auction_id);
+		let auction_balance = <<T as crate::Config>::Currency as Currency<T::AccountId>>::free_balance(auction_account);
+
+		if auction_balance.is_zero() {
+			destroy_auction_data = true;
+		}
+
+		Ok(destroy_auction_data)
+	}
+
+	///
+	/// Validates common and specific auction data
+	/// 
+	fn validate_data(&self) -> DispatchResult {
+		Pallet::<T>::validate_common_data(&self.common_data)?;
+
+		let default_duration = self
+			.common_data
+			.start
+			.checked_add(&T::BlockNumber::from(T::CandleDefaultDuration::get()))
+			.ok_or(Error::<T>::Overflow)?;
+
+		ensure!(
+			self.common_data.end == default_duration,
+			Error::<T>::CandleAuctionMustHaveDefaultDuration
+		);
+
+		ensure!(
+			self.common_data.reserve_price.is_none(),
+			Error::<T>::CandleAuctionDoesNotSupportReservePrice
+		);
+
+		let closing_period_duration = self
+			.common_data
+			.end
+			.checked_sub(&T::BlockNumber::from(T::CandleDefaultClosingPeriodDuration::get()))
+			.ok_or(Error::<T>::Overflow)?;
+
+		ensure!(
+			self.specific_data.closing_start == closing_period_duration,
+			Error::<T>::CandleAuctionMustHaveDefaultClosingPeriodDuration
+		);
+
+		Ok(())
+	}
+}
+
+impl<T: Config> Pallet<T> {
+  fn choose_random_block_from_range(from: u32, to: u32) -> Result<u32, DispatchError> {
+		ensure!(from < to, Error::<T>::InvalidTimeConfiguration);
+		let mut random_number = Self::generate_random_number(0u32);
+
+		let difference = to.checked_sub(from).ok_or(Error::<T>::Overflow)?;
```

Comment:
i tried using only sp_runtime::ArithmeticError for all math errors but i get the following error:

```^ the trait std::convert::From<sp_runtime::ArithmeticError> is not implemented for pallet::Error<T>```

weird because i look at Martin's Omnipool implementation and i did not see any specific impl for ArithmeticError.